src/deparse.c

```diff
@@ -0,0 +1,260 @@
+/*
+ * This file and its contents are licensed under the Apache License 2.0.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-APACHE for a copy of the license.
+ */
+
+#include <postgres.h>
+#include <utils/rel.h>
+#include <lib/stringinfo.h>
+#include <utils/builtins.h>
+#include <utils/lsyscache.h>
+#include <utils/relcache.h>
+#include <catalog/indexing.h>
+#include <utils/ruleutils.h>
+#include <utils/syscache.h>
+#include <commands/tablespace.h>
+#include <catalog/pg_class.h>
+#include <utils/rel.h>
+#include <access/relscan.h>
+#include <utils/fmgroids.h>
+#include <catalog/pg_constraint.h>
+#include <catalog/pg_index.h>
+#include <nodes/pg_list.h>
+#include "export.h"
+#include "compat.h"
+#include "deparse.h"
+
+/**
+ * Deparse a table into a set of SQL commands that can be used to recreate it.
+ * Together with column definiton it deparses constraints, indexes, triggers and rules as well.
+ * There are some table types that are not supported: temporary, partitioned,
+ * foreign, inherited and a table that uses options. Row security is also not supported.
+ */
+
+TS_FUNCTION_INFO_V1(ts_deparse_table);
+
+Datum
+ts_deparse_table(PG_FUNCTION_ARGS)
+{
+	Oid			relid = PG_GETARG_OID(0);
+
+	PG_RETURN_TEXT_P(cstring_to_text(deparse_table(relid)));
+}
+
+static void
+deparse_columns (StringInfo stmt, Relation rel)
+{
+	TupleConstr *constraints = rel->rd_att->constr;
+
+	for (int i = 0; i < rel->rd_att->natts; i++)
+	{
+		Form_pg_attribute attr = TupleDescAttr(rel->rd_att, i);
+
+		if (attr->attisdropped)
+			continue;
+
+		appendStringInfo(stmt, "\"%s\" %s",
+						 NameStr(attr->attname),
+						 format_type_with_typemod_qualified(attr->atttypid, attr->atttypmod));
+
+		if (attr->attnotnull)
+			appendStringInfoString(stmt, " NOT NULL");
+
+		if (OidIsValid(attr->attcollation))
+			appendStringInfo(stmt, " COLLATE \"%s\"",
+							 get_collation_name(attr->attcollation));
+
+		if (attr->atthasdef)
+		{
+			for (int i = 0; i < constraints->num_defval; i++)
```

Comment:
declaration in for loop.

For clarity I would also use another variable name in nested for loops so the index here to not override the previous `i`. For instance, use `j` here.